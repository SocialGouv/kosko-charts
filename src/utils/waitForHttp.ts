import type { IIoK8sApiCoreV1Container } from "kubernetes-models/_definitions/IoK8sApiCoreV1Container";

/** params for the [[waitForHttp]] function */
export interface WaitForHttpParams {
  /** name of the container */
  name: string;
  /** url to wait for */
  url: string;
}

/**
 * This creates a [[Container]] using the [wait-for-http](https://github.com/SocialGouv/docker/tree/master/wait-for-http) docker image
 *
 * ```typescript
 * import { waitForHttp } from "@socialgouv/kosko-charts/utils"
 *
 * const container = waitForHttp({
 *   name: "wait-for-free",
 *   url: "http://www.free.fr"
 * });
 * ```
 * @category utils
 * @return {IIoK8sApiCoreV1Container}
 */
export const waitForHttp = ({
  name,
  url,
}: WaitForHttpParams): IIoK8sApiCoreV1Container => {
  return {
    args: [url],
    image: `registry.gitlab.factory.social.gouv.fr/socialgouv/docker/wait-for-http:2.4.0`,
    imagePullPolicy: "Always",
    name: `wait-for-${name}`,
    resources: {
      limits: {
        cpu: "20m",
        memory: "32Mi",
      },
      requests: {
        cpu: "5m",
        memory: "16Mi",
      },
    },
  };
};
