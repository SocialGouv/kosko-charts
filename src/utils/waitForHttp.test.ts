import { waitForHttp } from "./waitForHttp";

test("should succeed validating a valid value", () => {
  expect(
    waitForHttp({ name: "some-service", url: "http://service:4242" })
  ).toMatchSnapshot();
});
