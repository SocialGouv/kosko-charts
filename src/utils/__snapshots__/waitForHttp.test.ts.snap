// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should succeed validating a valid value 1`] = `
Object {
  "args": Array [
    "http://service:4242",
  ],
  "image": "registry.gitlab.factory.social.gouv.fr/socialgouv/docker/wait-for-http:5.0.1",
  "imagePullPolicy": "Always",
  "name": "wait-for-some-service",
  "resources": Object {
    "limits": Object {
      "cpu": "20m",
      "memory": "32Mi",
    },
    "requests": Object {
      "cpu": "5m",
      "memory": "16Mi",
    },
  },
}
`;
